pkg/mcshub/controller/controller_test.go

```diff
@@ -97,411 +82,291 @@ func TestImportGenerated(t *testing.T) {
-	// maybe its not needed, and can be initialized
-	excepted := generateExceptedServiceImport(exp, es)
-	assertions := require.New(t)
-	// make sure that the right serviceImport was created
-	compareSi(&excepted, &si, assertions)
-}
-
-/*
-
-OLD VERSION - trying to use the client to create seperatlly
-
-func Test2ImportsFromExports(t *testing.T) {
-	exp1, es1 := prepareServiceExport(t)
-
-	preloadedObjects := []runtime.Object{service}
-	ser := mcscontroller.ServiceExportReconciler{
-		Client: getClient(preloadedObjects),
-		Log:    newLogger(t, false),
-		Scheme: getScheme(),
-	}
-	ser.Client.Create(context.TODO(), exp1)
-	req1 := reconcile.Request{
+	_, err = ser.Reconcile(context.TODO(), reconcile.Request{
 		NamespacedName: types.NamespacedName{
 			Name:      exp1.GetName(),
 			Namespace: exp1.GetNamespace(),
-		}}
-
-	// mimics the changing in the names that happens in the hub
-	// TO PR - make sure it suppose to be after taking the names to the req. maybe don't need it?
-	exp1.Name = lhutil.GenerateObjectName(serviceName, serviceNS, cluster1)
-	exp1.Namespace = brokerNS
-
-	result, err := ser.Reconcile(context.TODO(), req1)
-
-	if err != nil {
-		t.Error(err)
-	}
-
-	if result.Requeue {
-		t.Error("unexpected requeue")
-	}
-
-	si := mcsv1a1.ServiceImport{}
-	err = ser.Client.Get(context.TODO(), req1.NamespacedName, &si)
-
-	if err != nil {
-		t.Error(err)
-	}
-	assertions := require.New(t)
-	excepted := generateExceptedServiceImport(exp1, es1)
-	compareSi(&excepted, &si, assertions)
-
-	exp2, es2 := prepareServiceExport(t)
-	exp2.Labels[lhconst.LighthouseLabelSourceCluster] = cluster2
-
-	ser.Client.Create(context.TODO(), exp2)
-	req2 := reconcile.Request{
+		}})
+	assert.Nil(err)
+	_, err = ser.Reconcile(context.TODO(), reconcile.Request{
 		NamespacedName: types.NamespacedName{
 			Name:      exp2.GetName(),
 			Namespace: exp2.GetNamespace(),
-		}}
+		}})
+	assert.Nil(err)
 
-	// mimics the changing in the names that happens in the hub
-	//TO PR - make sure it suppose to be after taking the names to the req. maybe don't need it?
-	exp2.Name = lhutil.GenerateObjectName(serviceName, serviceNS, cluster2)
-	exp2.Namespace = brokerNS
-	exp2.Labels[lhconst.LighthouseLabelSourceCluster] = cluster2 // changed field
-	result, err = ser.Reconcile(context.TODO(), req2)
+	imports := &mcsv1a1.ServiceImportList{}
+	err = ser.Client.List(context.TODO(), imports, client.InNamespace(brokerNS))
+	assert.Nil(err)
 
-	if err != nil {
-		t.Error(err)
-	}
-	if result.Requeue {
-		t.Error("unexpected requeue")
-	}
+	assert.Len(imports.Items, 2, "expected 2 imports got", len(imports.Items))
 
-	err = ser.Client.Get(context.TODO(), req2.NamespacedName, &si)
+	exports := &mcsv1a1.ServiceExportList{}
+	err = ser.Client.List(context.TODO(), exports, client.InNamespace(brokerNS))
+	assert.Nil(err)
 
-	if err != nil {
-		t.Error(err)
+	for _, exp := range exports.Items {
+		conflict := lhutil.GetServiceExportCondition(&exp.Status, mcsv1a1.ServiceExportConflict)
```

Comment:
can refactor this into a function assertNoConflict() as you use it in multiple places.

Reply:
refactored into hasConflict(export, conflictStatus) since in some cases expect a conflict